server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -202,6 +204,21 @@ class InteractiveSession(
     }
   }
 
+  private def findSparkRArchives(): Seq[String] = {
+    sys.env.get("SPARKR_ARCHIVES_PATH")
+      .map(_.split(",").toSeq)
+      .getOrElse {
+        sys.env.get("SPARK_HOME").map { case sparkHome =>
+          val rLibPath = Seq(sparkHome, "R", "lib").mkString(File.separator)
+          val rArchivesFile = new File(rLibPath, "sparkr.zip")
+          require(rArchivesFile.exists(),
+            "sparkr.zip not found; cannot run sparkr application in YARN mode.")
```

Comment:
The message is misleading; this method is called regardless of how you're running Spark, so if the package is not found, you'll fail with this message even if the user has configured Spark to run in standalone mode.